### PR TITLE
Add Warehouse FAQ

### PR DESCRIPTION
**Pull request by ameypv-rudder**
## What do these changes do?

> Added warehouse FAQ on the automatic type conversions supported by RudderStack

## Type of documentation

- [ ] New documentation
- [x] Documentation update
- [ ] Hotfix

### CODE REVIEW

File: docs/destinations/warehouse-destinations/faq.mdx [hunk 1]
```diff
@@ -288,6 +288,21 @@ The table name truncation logic is as follows:
 - For **Data Lake** destinations (**S3 Data Lake, GCS Data Lake, Azure Data Lake**), there is no set limit.
 - For **other** destinations, the table name is truncated after 127 characters.
 
+### How does RudderStack handle schema data type mismatch when sending events to a warehouse destination?
+
+RudderStack automatically handles the data type conversion for some common scenarios **depending on the warehouse column data type**.
```
Comment (ameypv-rudder) [inline docs/destinations/warehouse-destinations/faq.mdx:293]:
```suggestion
RudderStack automatically handles the data type conversion in some common scenarios **depending on the warehouse column data type**.
```

File: docs/destinations/warehouse-destinations/faq.mdx [hunk 2]
```diff
@@ -288,6 +288,21 @@ The table name truncation logic is as follows:
 - For **Data Lake** destinations (**S3 Data Lake, GCS Data Lake, Azure Data Lake**), there is no set limit.
 - For **other** destinations, the table name is truncated after 127 characters.
 
+### How does RudderStack handle schema data type mismatch when sending events to a warehouse destination?
+
+RudderStack automatically handles the data type conversion in some common scenarios **depending on the warehouse column data type**.
+
+| Incoming data type | Warehouse column data type | Notes |
+| :----| :------| :------|
+| Integer <br /> Float <br /> JSON | String | RudderStack automatically stringifies the non-string data. |
```
Comment (ameypv-rudder) [inline docs/destinations/warehouse-destinations/faq.mdx:297]:
```suggestion
| All data types **except** String, Text, or JSON | String | RudderStack automatically stringifies the incoming data. |
```

File: docs/destinations/warehouse-destinations/faq.mdx [hunk 3]
```diff
@@ -288,6 +288,21 @@ The table name truncation logic is as follows:
...
+| Integer | Float | - |
```
Comment (ameypv-rudder) [inline docs/destinations/warehouse-destinations/faq.mdx:298]:
```suggestion
| Integer <br /> BigInt | Float | - | 
```

File: docs/destinations/warehouse-destinations/faq.mdx [hunk 4]
```diff
@@ -288,6 +288,21 @@ The table name truncation logic is as follows:
...
 
+### How does RudderStack handle schema data type mismatch when sending events to a warehouse destination?
+
+RudderStack automatically handles the data type conversion in some common scenarios **depending on the warehouse column data type**.
+
+| Incoming data type | Warehouse column data type | Notes |
+| :----| :------| :------|
+| Integer <br /> Float <br /> JSON | String | RudderStack automatically stringifies the non-string data. |
+| Integer | Float | - | 
+| Float | Integer | - |
```
Comment (ameypv-rudder) [inline docs/destinations/warehouse-destinations/faq.mdx:299]:
```suggestion
| Float | Integer <br /> BigInt | - |
```

File: docs/destinations/warehouse-destinations/faq.mdx [hunk 5]
```diff
@@ -288,6 +288,21 @@ The table name truncation logic is as follows:
 - For **Data Lake** destinations (**S3 Data Lake, GCS Data Lake, Azure Data Lake**), there is no set limit.
 - For **other** destinations, the table name is truncated after 127 characters.
 
+### How does RudderStack handle schema data type mismatch when sending events to a warehouse destination?
+
+RudderStack automatically handles the data type conversion in some common scenarios **depending on the warehouse column data type**.
+
+| Incoming data type | Warehouse column data type | Notes |
+| :----| :------| :------|
+| All data types **except** String, Text, or JSON | String | RudderStack automatically stringifies the incoming data. |
```
Comment (achettyiitr) [inline docs/destinations/warehouse-destinations/faq.mdx:297]:
```suggestion
| All data types **except** String, Text, or JSON | String or Text | RudderStack automatically stringifies the incoming data. |
```

**Review by achettyiitr — CHANGES_REQUESTED** [review]

**Review by achettyiitr — APPROVED** [review]

File: docs/destinations/warehouse-destinations/faq.mdx [hunk 6]
```diff
@@ -288,6 +288,21 @@ The table name truncation logic is as follows:
 - For **Data Lake** destinations (**S3 Data Lake, GCS Data Lake, Azure Data Lake**), there is no set limit.
 - For **other** destinations, the table name is truncated after 127 characters.
 
+### How does RudderStack handle schema data type mismatch when sending events to a warehouse destination?
+
+RudderStack automatically handles the data type conversion in some common scenarios **depending on the warehouse column data type**.
+
+| Incoming data type | Warehouse column data type | Notes |
```
Comment (ameypv-rudder) [inline docs/destinations/warehouse-destinations/faq.mdx:295]:
```suggestion
| Scenario | Incoming data type | Warehouse column data type | Notes |
```

File: docs/destinations/warehouse-destinations/faq.mdx [hunk 7]
```diff
@@ -288,6 +288,21 @@ The table name truncation logic is as follows:
...
+| :----| :------| :------|
```
Comment (ameypv-rudder) [inline docs/destinations/warehouse-destinations/faq.mdx:296]:
```suggestion
| :-----| :----| :------| :------|
```

File: docs/destinations/warehouse-destinations/faq.mdx [hunk 8]
```diff
@@ -288,6 +288,21 @@ The table name truncation logic is as follows:
...
+| All data types **except** String, Text, or JSON | String or Text | RudderStack automatically stringifies the incoming data. |
```
Comment (ameypv-rudder) [inline docs/destinations/warehouse-destinations/faq.mdx:297]:
```suggestion
| 1 | All data types **except** String, Text, or JSON | String or Text | RudderStack automatically stringifies the incoming data. |
```

File: docs/destinations/warehouse-destinations/faq.mdx [hunk 9]
```diff
@@ -288,6 +288,21 @@ The table name truncation logic is as follows:
...
+| Integer <br /> BigInt | Float | - |
```
Comment (ameypv-rudder) [inline docs/destinations/warehouse-destinations/faq.mdx:298]:
```suggestion
| 2 | Integer <br /> BigInt | Float | - | 
```

File: docs/destinations/warehouse-destinations/faq.mdx [hunk 10]
```diff
@@ -288,6 +288,21 @@ The table name truncation logic is as follows:
...
 
+### How does RudderStack handle schema data type mismatch when sending events to a warehouse destination?
+
+RudderStack automatically handles the data type conversion in some common scenarios **depending on the warehouse column data type**.
+
+| Incoming data type | Warehouse column data type | Notes |
+| :----| :------| :------|
+| All data types **except** String, Text, or JSON | String or Text | RudderStack automatically stringifies the incoming data. |
+| Integer <br /> BigInt | Float | - | 
+| Float | Integer <br /> BigInt | - |
```
Comment (ameypv-rudder) [inline docs/destinations/warehouse-destinations/faq.mdx:299]:
```suggestion
| 3 | Float | Integer <br /> BigInt | - |
```

File: docs/destinations/warehouse-destinations/faq.mdx [hunk 11]
```diff
@@ -288,6 +288,21 @@ The table name truncation logic is as follows:
...
+| Int <br /> Float <br /> Boolean | JSON | RudderStack automatically stringifies the incoming data. |
```
Comment (ameypv-rudder) [inline docs/destinations/warehouse-destinations/faq.mdx:300]:
```suggestion
| 4 | Integer <br /> Float <br /> Boolean | JSON | RudderStack automatically stringifies the incoming data. |
```

**Review by ishachopra0401 — APPROVED** [review]

**Comment by ishachopra0401** [comment]
LGTM! just checking if we can add rudder_discards scenario for the rest of the cases.

**Comment by achettyiitr** [comment]
> LGTM! just checking if we can add rudder_discards scenario for the rest of the cases.

@ishachopra0401 Good point. Anything which is not handled goes to rudder_discards.

**Comment by ameypv-rudder** [comment]
Thanks for the feedback @achettyiitr and @ishachopra0401 - I've incorporated the suggestion re: `rudder_discards`

File: docs/destinations/warehouse-destinations/faq.mdx [hunk 12]
```diff
@@ -288,6 +288,25 @@ The table name truncation logic is as follows:
...
+
+| Scenario | Incoming data type | Warehouse column data type | Notes |
+| :-----| :----| :------| :------|
+| 1 | All data types **except** String, Text, or JSON | String or Text | RudderStack automatically stringifies the incoming data. |
+| 2 | Integer <br /> BigInt | Float | - | 
+| 3 | Float | Integer <br /> BigInt | - |
+| 4 | Integer <br /> Float <br /> Boolean | JSON | RudderStack automatically stringifies the incoming data. |
+
+<div class="warningBlock">
+The incoming data which <strong>cannot</strong> be typecasted into the warehouse column-specific data type is sent to the <code class="inline-code">rudder_discards</code> table.
```
Comment (ishachopra0401) [inline docs/destinations/warehouse-destinations/faq.mdx:303]:
```suggestion
In any scenario other than the ones covered above, that is, when the incoming data <strong>cannot</strong> be typecasted into the warehouse column-specific data type, it is sent to the <code class="inline-code">rudder_discards</code> table.
```

**Review by ishachopra0401 — APPROVED** [review]

**Pull request merged**